blueocean-pipeline-editor/src/main/js/components/editor/AgentConfiguration.jsx

```diff
@@ -113,6 +114,22 @@ export class AgentConfiguration extends Component<DefaultProps, Props, State> {
         focusOnElement('.agent-select .required input');
     }
 
+    _getAgentInputControl(param, pristine, val) {
+        if (param.name === 'args') {
+            return <InputTextArea hasError={param.isRequired && !pristine && !val}
+                isRequired={param.isRequired}
+                defaultValue={val}
+                onChange={val => { this.setAgentValue(param.name, val); param.isRequired && this.setState({ pristine: false }); }}
+                onBlur={e => param.isRequired && this.setState({ pristine: false })} />
+        } else {
+            return <InputText hasError={param.isRequired && !pristine && !val}
+                isRequired={param.isRequired}
+                defaultValue={val}
+                onChange={val => { this.setAgentValue(param.name, val); param.isRequired && this.setState({ pristine: false }); }}
+                onBlur={e => param.isRequired && this.setState({ pristine: false })} />
```

Comment:
Is there a more elegant way to avoid all of the duplicated props here? Could we create an intermediate object and then explode those props onto the desired component using the spread operator?

```
const inputProps = {
   hasError: param.isRequired && !pristine && !val,
   // continue other props
}
```

and then

```
if (param.name === 'args') {
    return <InputTextArea {...inputProps} />
} else {
   return <InputText {...inputProps} />
}
```

It might not be quite _that_ simple, need to watch out for the more complex function bindings there... but to me this kind of code dup is a 🐛  unless there is literally no other way to make it work.